### ISSUE TITLE
KB2686509 fails to install if Windows Embedded POSReady 2009 updates are enabled (Windows XP SP3)

### ISSUE THREAD

**Issue by Lost-Entrepreneur439** [kirb/LegacyUpdate#164]
If Embedded POSReady 2009 updates are enabled, KB2686509 will fail to install, as Legacy Update will attempt to install the Windows XP Embedded version of it, which is not compatible with standard Windows XP, if you grab the XP Embedded KB2686509 update from the update catalog and attempt to install it, it'll give you the error "Setup cannot continue because one or more prerequisites required to install KB2686509 failed. For More details check the Log File c:\windows\KB2686509.log", however since KB2686509 is an update from 2012, it has a standard XP version, and if you grab that from the update catalog, it'll install successfully. Legacy Update should be installing the standard version, not the embedded version. The log is attached (the last part of the log is from when I installed the standard XP version)
[KB2686509.log](https://github.com/kirb/LegacyUpdate/files/12210218/KB2686509.log)

**Comment by sal-ammoniac** [comment]
This problem also exists in official WU. I memorized that before WU v6 shutdown,I also find this issue as it was always failing.So removing the POSReady registry or pre-installing KB2686529 may solve it?

**Comment by PandemicOfNukes** [comment]
yes to solve this problem I pre-installed KB2686529 (non embedded version) from the windows catalog before installing Legacy Update.

Here is the 32-bit version: https://www.catalog.update.microsoft.com/ScopedViewInline.aspx?updateid=77643db3-5759-497a-82ac-60f111f6f43b
 
And 64-bit version: https://www.catalog.update.microsoft.com/ScopedViewInline.aspx?updateid=7455cd67-9ba9-48f8-8d71-959675411ed4

**Comment by kirb** [comment]
Decided to address this by faking this update as already installed while enabling Embedded 2009 updates in the registry.

KB2686509, in my opinion, has no value. It doesn’t fix anything - it just one-shot creates a log file containing keyboard layouts in the registry that are vulnerable to [dll sideloading attacks](https://attack.mitre.org/techniques/T1574/001/). The value of this is very questionable, because the root issue is that the registry on a standard install of Windows contains thousands of references to system files that are a bare name (e.g. `kbdus.dll`) rather than a fully-qualified path (e.g. `C:\Windows\system32\kbdus.dll`). Even if this update did in fact fix the keyboard layout registry keys, there are many more parts of the registry that remain vulnerable.

Basically: Windows is just inherently vulnerable to this attack, and it remains barely fixed. Each program needs to [opt into](https://github.com/LegacyUpdate/LegacyUpdate/blob/c895a45/shared/Startup.h#L8) using more secure dll search algorithms, and/or to ensure they only ever call `LoadLibrary()` with fully-qualified paths. Even the registry keys this update checks for remain unfixed in Windows 11 25H2, although it’s possible whichever dll loads the layout dlls prepends system32 to the bare dll filenames (an actual fix!).

It’s described in some more detail in this mailing list post: https://seclists.org/fulldisclosure/2013/Oct/151

Note that this won’t change anything for users who have already enabled Embedded 2009 updates, because the checkbox for it won’t be shown. It will also still be offered, and correctly “install”, on Windows XP (without Embedded 2009 updates), Server 2003, and Embedded 2009.
